plaid/client.py

```diff
@@ -33,25 +33,31 @@ def __init__(self,
                  public_key,
                  environment,
                  suppress_warnings=False,
-                 timeout=DEFAULT_TIMEOUT):
+                 timeout=DEFAULT_TIMEOUT,
+                 http_client=post_request,
+                 use_async=False):
         '''
         Initialize a client with credentials.
 
-        :param  str     client_id:          Your Plaid client ID
-        :arg    str     secret:             Your Plaid secret
-        :arg    str     public_key:         Your Plaid public key
-        :arg    str     environment:        One of ``sandbox``,
+        :param  str      client_id:          Your Plaid client ID
+        :arg    str      secret:             Your Plaid secret
+        :arg    str      public_key:         Your Plaid public key
+        :arg    str      environment:        One of ``sandbox``,
                                             ``development``, or ``production``.
-        :arg    bool    suppress_warnings:  Suppress Plaid warnings.
-        :arg    int     timeout:            Timeout for API requests.
-
+        :arg    bool     suppress_warnings:  Suppress Plaid warnings.
+        :arg    int      timeout:            Timeout for API requests.
+        :arg    callable http_client:        Request adaptor. sync or async
+        :arg    bool     use_async:          Whether or not to use
```

Comment:
a nit on the function signature: since we're already adding the `http_client` argument to make it possible to plug in `post_request_async`, I'd prefer to avoid introducing a second method of specifying async-ness by also adding the `use_async` argument.